pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -561,3 +564,34 @@ func (c *Cluster) resolveNode(nodeName string) *rookalpha.Node {
 	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, c.resources)
 	return rookNode
 }
+
+func (c *Cluster) getValidStorageNodes() []rookalpha.Node {
+	storageNodes := c.Storage.Nodes
+	nodeOptions := metav1.ListOptions{}
+	nodeOptions.TypeMeta.Kind = "Node"
+	allNodes, err := c.context.Clientset.CoreV1().Nodes().List(nodeOptions)
+	if err != nil {
+		// cannot list nodes, assume the specified nodes are valid
+		logger.Warningf("failed to list nodes: %v", err)
+		return storageNodes
+	}
+	storageNodeNames := util.NewSet()
+	for _, storageNode := range storageNodes {
+		storageNodeNames.Add(storageNode.Name)
+	}
+	validNodes := []rookalpha.Node{}
+	for _, node := range allNodes.Items {
+		if storageNodeNames.Contains(node.Name) {
+			valid, err := k8sutil.ValidNode(node, c.placement)
+			if err != nil {
+				logger.Warning("failed to validate node %v", err)
```

Comment:
there are several of these "failed to validate node" messages where it would be nice to include the node name.

Reply:
done